cljfmt/src/cljfmt/core.cljc

```diff
@@ -681,6 +687,43 @@
         aligned? #(aligned-form? % aligned-forms context)]
     (transform form edit-all aligned? align-columns)))
 
+(defn- blank-line-rule [zloc blank-line-forms context]
+  (when-let [form-symb (form-symbol zloc)]
+    (some blank-line-forms
+          [(remove-namespace form-symb)
+           (fully-qualified-symbol form-symb context)])))
+
+(defn- blank-line-in-form? [zloc blank-line-forms context]
+  (and (z/linebreak? zloc)
+       (> (count-newlines zloc) 1)
+       (cond
+         (z/list? (z/up zloc))
+         (not= (blank-line-rule zloc blank-line-forms context) :all)
+
+         (z/list? (z/up (z/up zloc)))
+         (let [index           (dec (index-of (z/up zloc)))
+               rule            (blank-line-rule
+                                (z/up zloc)
+                                blank-line-forms
+                                context)
+               allowed-indexes (if (set? rule)
+                                 rule
+                                 #{})]
+           (not (allowed-indexes index))))))
```

Comment:
This doesn't seem like it'll work for an arbitrary nesting depth. What we want to do is determine whether the zloc represents multiple newlines, and then iterate up the tree until we hit root (`nil`), checking if any containing form is also a matching form.